src/actions/App.actions.js

```diff
@@ -1,53 +1,67 @@
 import { ethers } from "ethers";
 import { addresses, Actions } from "../constants";
-import { abi as ierc20Abi } from "../abi/IERC20.json";
-import { abi as OHMPreSale } from "../abi/OHMPreSale.json";
 import { abi as OlympusStaking } from "../abi/OlympusStaking.json";
 import { abi as OlympusStakingv2 } from "../abi/OlympusStakingv2.json";
-import { abi as MigrateToOHM } from "../abi/MigrateToOHM.json";
 import { abi as sOHM } from "../abi/sOHM.json";
 import { abi as sOHMv2 } from "../abi/sOhmv2.json";
-import { abi as LPStaking } from "../abi/LPStaking.json";
-import { abi as DistributorContract } from "../abi/DistributorContract.json";
-import { abi as BondContract } from "../abi/BondContract.json";
-import { abi as DaiBondContract } from "../abi/DaiBondContract.json";
-import { abi as PairContract } from "../abi/PairContract.json";
-import { abi as CirculatingSupplyContract } from "../abi/CirculatingSupplyContract.json";
 import axios from 'axios';
 import { contractForReserve, addressForAsset } from "../helpers";
 import { BONDS } from "../constants";
 import { abi as BondOhmDaiCalcContract } from "../abi/bonds/OhmDaiCalcContract.json";
-
-const parseEther = ethers.utils.parseEther;
+import client from "../lib/apolloClient.js";
+import { gql } from "@apollo/client";
+
+const protocolMetricsQuery = `
+  query {
+    _meta {
+      block {
+        number
+      }
+    }
+    protocolMetrics(first: 1, orderBy: timestamp, orderDirection: desc) {
+      timestamp
+      circulatingSupply
+      totalSupply
+      ohmPrice
+      marketCap
+      totalValueLocked
+    }
+  }
+`;
 
 export const fetchAppSuccess = payload => ({
   type: Actions.FETCH_APP_SUCCESS,
   payload,
 });
 
-async function calculateAPY(sohmContract, stakingReward) {
-  const circSupply = await sohmContract.circulatingSupply();
-
-  const stakingRebase = stakingReward / circSupply;
-  const stakingAPY = Math.pow(1 + stakingRebase, 365 * 3) - 1;
-
-  return stakingAPY;
-}
 
 export const loadAppDetails =
   ({ networkID, provider }) =>
   async dispatch => {
-    const currentBlock = await provider.getBlockNumber();
 
-    const ohmContract = new ethers.Contract(addresses[networkID].OHM_ADDRESS, ierc20Abi, provider);
+    const graphData = await client.query({
+      query: gql(protocolMetricsQuery)
+    })
+    .then(data => {
```

Comment:
Since you're using `await` above, it'll block execution until the client returns. So `then `isn't necessary here